src/main/java/net/pms/newgui/SharedContentTab.java

```diff
@@ -0,0 +1,748 @@
+
+		JScrollPane pane = new JScrollPane(webContentList);
+		Dimension d = webContentList.getPreferredSize();
+		pane.setPreferredSize(new Dimension(d.width, webContentList.getRowHeight() * 2));
+		builderFolder.add(pane, FormLayoutUtil.flip(cc.xyw(1, 5, 7), colSpec, orientation));
+
+		return builderFolder;
+	}
+
+	public static void setScanLibraryEnabled(boolean enabled) {
+		scanButton.setEnabled(enabled);
+		scanButton.setIcon(scanNormalIcon);
+		scanButton.setRolloverIcon(scanRolloverIcon);
+		scanButton.setPressedIcon(scanPressedIcon);
+		scanButton.setDisabledIcon(scanDisabledIcon);
+		scanButton.setToolTipText(Messages.getString("FoldTab.2"));
+	}
+
+	/**
+	 * @todo combine with setScanLibraryEnabled after we are in sync with DMS
+	 */
+	public static void setScanLibraryBusy() {
+		scanButton.setIcon(scanBusyIcon);
+		scanButton.setRolloverIcon(scanBusyRolloverIcon);
+		scanButton.setPressedIcon(scanBusyPressedIcon);
+		scanButton.setDisabledIcon(scanBusyDisabledIcon);
+		scanButton.setToolTipText(Messages.getString("FoldTab.40"));
+	}
+
+	public class SharedFoldersTableModel extends DefaultTableModel {
+		private static final long serialVersionUID = -4247839506937958655L;
+
+		public SharedFoldersTableModel() {
+			super(new String[]{Messages.getString("FoldTab.56"), Messages.getString("FoldTab.65")}, 0);
+		}
+
+		@Override
+		public Class<?> getColumnClass(int columnIndex) {
+			Class clazz = String.class;
+			switch (columnIndex) {
+				case 1:
+					clazz = Boolean.class;
+					break;
+				default:
+					break;
+			}
+			return clazz;
+		}
+
+		@Override
+		public boolean isCellEditable(int row, int column) {
+			return column == 1;
+		}
+
+		@Override
+		public void setValueAt(Object aValue, int row, int column) {
+			Vector rowVector = (Vector) dataVector.elementAt(row);
+			if (aValue instanceof Boolean && column == 1) {
+				rowVector.setElementAt((boolean) aValue, 1);
+			} else {
+				rowVector.setElementAt(aValue, column);
+			}
+			fireTableCellUpdated(row, column);
+			updateSharedFoldersModel();
+		}
+	}
+
+	public class WebContentTableModel extends DefaultTableModel {
+		private static final long serialVersionUID = -4247839506937958655L;
+
+		public WebContentTableModel() {
+			// Column headings
+			super(new String[]{
+				Messages.getString("SharedContentTab.Type"),
+				Messages.getString("SharedContentTab.FolderName"),
+				Messages.getString("SharedContentTab.Source"),
+			}, 0);
+		}
+
+		@Override
+		public Class<?> getColumnClass(int columnIndex) {
+			return String.class;
+		}
+
+		@Override
+		public boolean isCellEditable(int row, int column) {
+			return false;
+		}
+
+		@Override
+		public void setValueAt(Object aValue, int row, int column) {
+			Vector rowVector = (Vector) dataVector.elementAt(row);
+			rowVector.setElementAt(aValue, column);
+			fireTableCellUpdated(row, column);
+			updateWebContentModel();
+		}
+	}
+
+	public class TableMouseListener extends MouseAdapter {
```

Comment:
@SubJunk you implemeted this listener and will you continue to add the feature when the mouse double click on the line will open the content panel to be able to edit the line setting? Now when user added a new line with wrong setting the only possibility is delete this wrong line and make the new one without possibility to edit them.

Reply:
@valib yeah that sounds like a good idea, will do that